decoder.go

```diff
@@ -517,37 +517,39 @@ func (d *decoder) decodeStruct(name string, node ast.Node, result reflect.Value)
 		structType := structVal.Type()
 		for i := 0; i < structType.NumField(); i++ {
 			fieldType := structType.Field(i)
+			tagParts := strings.Split(fieldType.Tag.Get(tagName), ",")
```

Comment:
Hi @mirdhyn 

Thank you for the PR. I think this might be a little cleaner if we reverse the logical operator here and just `continue` if the tag is `-`, for example:

```go
tagParts := strings.Split(fieldType.Tag.Get(tagName), ",")

// Ignore fields with tag name "-"
if tagParts[0] == "-" {
  continue
}

if fieldType.Anonymous {
 // ...
}
```

This will save us the additional layer of indentation, and I think it has the same result. What do you think?